### PR TITLE
feat: add neutralLoss method (issue #236)

### PR DESCRIPTION
**Pull request by jorainer**
- Add `neutralLoss` method.
- Add `PrecursorMzParam` to implement the precursor m/z - fragment m/z neutral
  loss calculation.

Also feedback on naming of the parameter object (`PrecursorMzParam` for the approach to calculate neutral loss spectra by precursor m/z - fragment m/z) welcome.

### CODE REVIEW

**Comment by jorainer** [comment]
@michaelwitting, can you please also have a look at this PR to check if this would work for you?

**Comment by michaelwitting** [comment]
Looks good to me so far, need to test with some own data knowning to work with that.

File: R/Spectra-neutralLoss.R [hunk 1]
```diff
@@ -0,0 +1,167 @@
+setClassUnion("functionOrNull", c("function", "NULL"))
+
+#' @title Calculate Neutral Loss Spectra
+#'
+#' @name neutralLoss
+#'
+#' @description
+#'
+#' This help page lists functions to convert MS/MS spectra to neutral loss
```
Comment (lgatto) [inline R/Spectra-neutralLoss.R:9]: functions that concert

File: R/Spectra-neutralLoss.R [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+#' spectra. The main function for this is `neutralLoss` and the specific
+#' algorithm to be used is defined (and configured) with dedicated *parameter*
+#' objects (paramer `param` of the `neutralLoss` function.
```
Comment (lgatto) [inline R/Spectra-neutralLoss.R:12]: (argument `param` of the `neutralLoss` function).

File: R/Spectra-neutralLoss.R [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+#'
+#' The parameter objects for the different algorithms are:
+#'
+#' - `PrecursorMzParam`: calculates neutral loss spectra as in Aisporna 2022 by
```
Comment (lgatto) [inline R/Spectra-neutralLoss.R:16]: as in Aisporna *et al.* (2022)

File: R/Spectra-neutralLoss.R [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+#'   subtracting the (fragment's) peak m/z value from the precursor m/z value
+#'   of each spectrum (precursor m/z - fragment m/z). Parameter `msLevel`
+#'   allows to restrict calculation of neutral loss spectra to specified MS
+#'   level(s). Spectra from other MS level(s) are returned as-is. Parameter
+#'   `filterPeaks` allows in addition to filter fragment peaks e.g. if their
```
Comment (lgatto) [inline R/Spectra-neutralLoss.R:21]: Not sure what the 'in addtion' means here. If it's in addition to the other parameters, I would drop it. If it's supposed to mean it has multiple effects, then I'm not sure what they are.

**Review by lgatto — CHANGES_REQUESTED** [review]

File: R/Spectra-neutralLoss.R (hunk 4 above)
Comment (jorainer) [reply R/Spectra-neutralLoss.R:21]: You're right. Better drop it.

**Comment by jorainer** [comment]
Changes done @lgatto

**Comment by jorainer** [comment]
Now I'm a little confused about the failing GHA. Locally all runs fine.

**Comment by jorainer** [comment]
OK, unit tests fixed. These were related to the recent change in `MsCoreUtils` (return a positive ppm, even for negative m/z)

**Review by lgatto — APPROVED** [review]

**Pull request merged**